influxdb.conf

```diff
@@ -1,434 +1,7 @@
-  #   # Default template
-  #   "server.*",
-  # ]
-
-###
-### [collectd]
-###
-### Controls one or many listeners for collectd data.
-###
-
-# [[collectd]]
-  # enabled = false
-  # bind-address = ":25826"
-  # database = "collectd"
-  # retention-policy = ""
-  # typesdb = "/usr/share/collectd/types.db"
-
-  # These next lines control how batching works. You should have this enabled
-  # otherwise you could get dropped metrics or poor performance. Batching
-  # will buffer points in memory if you have many coming in.
-
-  # Flush if this many points get buffered
-  # batch-size = 5000
-
-  # Number of batches that may be pending in memory
-  # batch-pending = 10
-
-  # Flush at least this often even if we haven't hit buffer limit
-  # batch-timeout = "10s"
-
-  # UDP Read buffer size, 0 means OS default. UDP listener will fail if set above OS max.
-  # read-buffer = 0
-
-###
-### [opentsdb]
-###
-### Controls one or many listeners for OpenTSDB data.
-###
-
-# [[opentsdb]]
-  # enabled = false
-  # bind-address = ":4242"
-  # database = "opentsdb"
-  # retention-policy = ""
-  # consistency-level = "one"
-  # tls-enabled = false
-  # certificate= "/etc/ssl/influxdb.pem"
-
-  # Log an error for every malformed point.
-  # log-point-errors = true
-
-  # These next lines control how batching works. You should have this enabled
-  # otherwise you could get dropped metrics or poor performance. Only points
-  # metrics received over the telnet protocol undergo batching.
-
-  # Flush if this many points get buffered
-  # batch-size = 1000
-
-  # Number of batches that may be pending in memory
-  # batch-pending = 5
-
-  # Flush at least this often even if we haven't hit buffer limit
-  # batch-timeout = "1s"
-
-###
-### [[udp]]
-###
-### Controls the listeners for InfluxDB line protocol data via UDP.
-###
-
-# [[udp]]
-  # enabled = false
-  # bind-address = ":8089"
-  # database = "udp"
-  # retention-policy = ""
-
-  # These next lines control how batching works. You should have this enabled
-  # otherwise you could get dropped metrics or poor performance. Batching
-  # will buffer points in memory if you have many coming in.
-
-  # Flush if this many points get buffered
-  # batch-size = 5000
-
-  # Number of batches that may be pending in memory
-  # batch-pending = 10
-
-  # Will flush at least this often even if we haven't hit buffer limit
-  # batch-timeout = "1s"
-
-  # UDP Read buffer size, 0 means OS default. UDP listener will fail if set above OS max.
-  # read-buffer = 0
-
-###
-### [continuous_queries]
-###
-### Controls how continuous queries are run within InfluxDB.
-###
-
-# [continuous_queries]
-  # Determiens whether the continuous query service is enabled.
-  # enabled = true
-
-  # Controls whether queries are logged when executed by the CQ service.
-  # log-enabled = true
-
-  # interval for how often continuous queries will be checked if they need to run
-  # run-interval = "1s"
-
+  dir = "/var/lib/influxdb/data"
```

Comment:
I would suggest maintaining backwards-compatibility so that users can just update their images without breaking anything, or having to re-configure again.